src/managed_data/client.py

```diff
@@ -123,13 +157,24 @@ def invoke(
     ) -> Union[dict, list]:
         """core call to API endpoint"""
 
+        url = urljoin(self.api_url, endpoint)
+
         response = requests.post(
-            urljoin(self.api_url, endpoint),
+            url,
             headers=self.headers,
             json=data,
         )
 
-        return _check_response(response)
+        try:
+            response = _check_response(response)
+        except Exception as e:
+            print(f"URL: {url}")
+            print(f"headers: {self.headers}")
+            print(f"data: {data}")
+
+            raise e
+
+        return response
```

Comment:
better error reporting makes it easier for us to debug when things go wrong. we can now reconstruct failing requests